internal/services/gcp_reservation_service.go

```diff
@@ -0,0 +1,139 @@
+package services
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/go-chi/render"
+	"github.com/lzap/dejq"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	"github.com/RHEnVision/provisioning-backend/internal/jobs"
+	"github.com/RHEnVision/provisioning-backend/internal/jobs/queue"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/RHEnVision/provisioning-backend/internal/payloads"
+)
+
+func CreateGCPReservation(w http.ResponseWriter, r *http.Request) {
+	logger := ctxval.Logger(r.Context())
+
+	var accountId int64 = ctxval.AccountId(r.Context())
+
+	// TODO: meanwhile these values are hardcoded until we will have instance types, sources, ssh endpoints,
+	payload := &payloads.GCPReservationRequestPayload{
+		PubkeyID: 2,
+		// TODO: The project id change to source id from sources
+		SourceID:    "citric-expanse-361512",
```

Comment:
I don't understand why we name a string which is obviously GCP Project ID as "SourceID", isn' this a typo? I would expect this to be named ProjectID instead. Even if sources name it badly, let's if this on our end.

Reply:
Project Id will be kept as a source, we will fetch that from sources.

Reply:
so we will keep the source id and will fetch the project id like we did in was and arn.

Reply:
Ah right! TY